File: server/game/lib/marriage.cpp
#include <fb/game/marriage.h>
#include <fb/protocol/flatbuffer/protocol.h>

namespace fb::game {

fb::protocol::internal::Marriage marriage::to_protocol() const
{
    return fb::protocol::internal::Marriage{spouse_id,
                                            spouse_name,
                                            fb::model::datetime() >= remarriage_after ? ""
                                                                                      : remarriage_after.to_string(),
                                            divorce_count};
}

} // namespace fb::game

File: server/http/Model/Marriage.cs
using StackExchange.Redis;

namespace Http.Model
{
    public class MarriageKey : BaseModel, IRedisValueKey
    {
        public required uint CharacterId { get; set; }

        public uint? GetHash() => CharacterId;

        public RedisKey GetRedisKey() => $"cache:marriage:{CharacterId}";
    }

    public class Marriage : MarriageKey, IModel
    {
        public uint? SpouseId { get; set; }

        public DateTime RemarriageAfter { get; set; } = DateTime.UtcNow;

        public uint DivorceCount { get; set; }
    }
}